### DISCUSSION TITLE
Table formatting and embedding media files

### DISCUSSION THREAD

**Discussion by jin-ahn** [zk-org/zk#133] [Help] [answered]
Hi I have a few questions on ZK that i wasn't able to figure from my testing:

1. Are there any short cuts for table formating? Rather than having to type out --- and | and manually space things out 
2. Is there support for attachments such as pdfs and jpegs? I know rendering these on the terminal is whole another things, but I was hoping to be able to at least link to them and search the file names of them?

**Comment by mickael-menu** [comment, marked as answer]
> * Are there any short cuts for table formating? Rather than having to type out --- and | and manually space things out

`zk` is not a note editor, only a CLI tool (with some ability to augment your text editor with the LSP server). So you will need to see how to handle that in your editor.

Personally on Neovim I'm using [vim-table-mode](https://github.com/dhruvasagar/vim-table-mode) for this particular use case.

> * Is there support for attachments such as pdfs and jpegs? I know rendering these on the terminal is whole another things, but I was hoping to be able to at least link to them and search the file names of them?

Not yet but being able to link to images would be quite useful indeed.
If you mean searching the filename occurrences in your notes you can use:

```sh
zk list --exact-match --match image-filename
# or with short flags
zk list -em image-filename
```

**Comment by srid** [comment]
Editing tables in Markdown is not fun especially when you want to use non-trivial markdown in the cells. I use the `list-table.lua` Pandoc filter to avoid tables entirely (but still render them as one). 

Example,

- Source (lists): https://raw.githubusercontent.com/srid/emanote/master/docs/demo/lua-filters.md
- Output (table): https://emanote.srid.ca/demo/lua-filters
